infra/main.bicep

```diff
@@ -101,11 +101,13 @@ param embeddingModelName string = ''
 param embeddingDeploymentName string = ''
 param embeddingDeploymentVersion string = ''
 param embeddingDeploymentCapacity int = 0
+param embeddingDimensions int = 0
 var embedding = {
   modelName: !empty(embeddingModelName) ? embeddingModelName : 'text-embedding-ada-002'
   deploymentName: !empty(embeddingDeploymentName) ? embeddingDeploymentName : 'embedding'
   deploymentVersion: !empty(embeddingDeploymentVersion) ? embeddingDeploymentVersion : '2'
   deploymentCapacity: embeddingDeploymentCapacity != 0 ? embeddingDeploymentCapacity : 30
+  dimensions: embeddingDimensions != 0 ? embeddingDimensions : 1536
```

Comment:
will this work on ada002 as well?

Reply:
I am doing more tests and am now getting an error with the old environments. Apparently the OpenAI API won't let us specify dimensions for ada002 models at all, even if its set to 1536. So I have to optionally include in all our create() calls only for the newer models.
Not sure why I didn't catch that before. I think I keep forgetting to delete md5's when switching environments.

Reply:
Made changes, tested in several old environments, both Azure and non-Azure.